### PR TITLE
Prevent adding null values to log entry labels.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
Continuation of #2837.
@henkmollema take a look and feel free to leave comments and suggestions, don't know why I can't add you as a reviewer.

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
@henkmollema Can you confirm here that you agree with this PR reusing your original commits so I can turn the CLA bit on. Thanks!

**Comment by henkmollema** [comment]
Yes I'm ok with that!

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/Logging/GoogleLoggerTest.cs [hunk 1]
```diff
@@ -359,6 +360,19 @@ public void Log_Labels()
             mockConsumer.VerifyAll();
         }
 
+        [Fact]
+        public void Log_DoesNotLogIfNullLabels()
```
Comment (henkmollema) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/Logging/GoogleLoggerTest.cs]: Wondering if there should be a test with a `RetryOptions.ExceptionHandling` value different than  `ExceptionHandling.Ignore` where the exception would propagated.

**Comment by jskeet** [comment]
Hi @henkmollema, Amanda is now on leave, so I'll manage this going forward. I'll admit I haven't looked into this whole area nearly as much as Amanda.
Adding tests is always nice, and I'm quite happy to do that - but are you okay with me doing it in a new PR? This is already sufficiently complicated in terms of being one PR migrated to another etc that I'd prefer to "start fresh" for the extra change. If that sounds okay to you, I'll merge this and work on the new test.

**Comment by jskeet** [comment]
CLA override reason: henkmollema has confirmed within this PR that he's okay using the code here.

**Comment by henkmollema** [comment]
Regarding the tests: fine by me to add them in a separate PR. 👍

**Comment by jskeet** [comment]
Thanks! Merging now.

**Comment by henkmollema** [comment]
Thanks Jon!

**Pull request merged**